Cython/Compiler/FusedNode.py

```diff
@@ -303,9 +304,19 @@ def _fused_instance_checks(self, normal_types, pyx_code, env):
                 """)
 
     def _dtype_name(self, dtype):
+        # avoid duplicating names
+        if dtype in self.used_names:
+            return self.used_names[dtype]
         if dtype.is_typedef:
-            return '___pyx_%s' % dtype
-        return str(dtype).replace(' ', '_')
+            base_name = ('___pyx_%s' % dtype).replace(' ', '_')
```

Comment:
I'm not sure if prepending just `__pyx_` is a good idea. There are lots of special names with that prefix, so it seems easy to trigger a naming collision with internal names from user code. We should use a dedicated namespace prefix for these.

Reply:
I think it's OK in that it's (almost) all just used in Cython code as local variables within the dispatch function. So they're all put through the standard variable name mangling anyway. But I've replace it with something more unique.